waiter/integration/waiter/token_watch_integration_test.clj

```diff
@@ -0,0 +1,362 @@
+          (try
+            (let [response (post-token waiter-url (assoc (kitchen-params) :token token-name) :cookies cookies)
+                  entry (get-token-index waiter-url token-name :cookies cookies)]
+              (assert-response-status response 200)
+              (assert-watches-token-index-entry watches token-name entry))
+            (let [response (post-token waiter-url (assoc (kitchen-params) :token token-name
+                                                                          :version "updated-version")
+                                       :cookies cookies)
+                  entry (get-token-index waiter-url token-name :cookies cookies)]
+              (assert-response-status response 200)
+              (assert-watches-token-index-entry watches token-name entry))
+            (delete-token-and-assert waiter-url token-name :hard-delete false)
+            (let [entry (get-token-index waiter-url token-name :cookies cookies)]
+              (assert-watches-token-index-entry watches token-name entry))
+            (finally
+              (stop-watches watches)
+              (delete-token-and-assert waiter-url token-name)))))
+
+      (testing "stream receives DELETE events for all routers"
+        (let [token-name (create-token-name waiter-url ".")
+              response (post-token waiter-url (assoc (kitchen-params) :token token-name) :cookies cookies)
+              watches (start-watches router-urls cookies)]
+          (assert-response-status response 200)
+          (try
+            (let [entry (get-token-index waiter-url token-name :cookies cookies)]
+              (assert-watches-token-index-entry watches token-name entry)
+              (delete-token-and-assert waiter-url token-name)
+              (assert-watches-token-index-entry watches token-name nil))
+            (finally
+              (stop-watches watches)))))
+
+      (testing "stream does not include soft deleted token events without include=deleted query param"
+        (let [token-1 (create-token-name waiter-url ".")
+              token-2 (create-token-name waiter-url ".")
+              res-1 (post-token waiter-url (assoc (kitchen-params) :token token-1) :cookies cookies)
+              res-2 (post-token waiter-url (assoc (kitchen-params) :token token-2) :cookies cookies)]
+          (assert-response-status res-1 200)
+          (assert-response-status res-2 200)
+          (try
+            (delete-token-and-assert waiter-url token-1 :hard-delete false)
+            (let [entry (get-token-index waiter-url token-2 :cookies cookies)
+                  watch (start-watch waiter-url cookies :query-params {"include" "metadata" "watch" "true"})]
+              (assert-watch-token-index-entry watch token-1 nil)
+              (assert-watch-token-index-entry watch token-2 entry)
+              (delete-token-and-assert waiter-url token-2 :hard-delete false)
+              (async/<!! (async/timeout 4000))
```

Comment:
Why do we have these sleeps?

Reply:
The server can possibly send an event during the `delete-token-and-assert`. If I do the assertion too soon (without the sleeps), it might be a false positive as the event can be process after the assertion.

Reply:
Can we use `wait-for` instead?

Reply:
I could assert that wati-for returns false